pkg/kubelet/deviceplugin/device_plugin_stub.go

```diff
@@ -67,14 +67,6 @@ func (m *Stub) Start() error {
 	pluginapi.RegisterDevicePluginServer(m.server, m)
 
 	go m.server.Serve(sock)
-	// Wait till grpc server is ready.
-	for i := 0; i < 10; i++ {
-		services := m.server.GetServiceInfo()
```

Comment:
Will removing this cause problems? e.g. if server is not ready, then some calls on it may fail.

Reply:
indeed. But doing this is incorrect as it just wait for 10s all the time. GRPC has a `FailFast` toggle that we can set to false to allow the connection to be retried. Usually the toggle is set to not fail fast as part of health check process ( that we can also use in test ) to determine if the server is responding within a time period.
But the current API doesn't expose the healthcheck API that we can set `grpc.FailFast(false)` on. Not sure if we should add one as part if this change.